test_fms/time_manager/test_time_manager.F90

```diff
@@ -680,6 +686,60 @@ program test_time_manager
     enddo
     write(outunit,'(a)') 'set_date_gregorian and get_date_gregorian tests successful'
  endif
+  !! mixed precision tests
+  if(test21) then
+    !! check r4 overloads from original real_to_time_type test
+    write(outunit,'(/,a)') '#################################  test21  #################################'
+    call print_time(real_to_time_type(real(86401.1, r4_kind)), 'real_to_time_type(86401.1):', unit=outunit)
+    Time = real_to_time_type(real(-1.0, r4_kind), err_msg)
+    if(err_msg == '') then
+       call mpp_error(FATAL, 'test21 fails: did not get the expected error message')
+    else
+      write(outunit,'(a)') 'test successful: '//trim(err_msg)
+    endif
+    !! check equality between r4 and r8 returned types
+    if ( real_to_time_type(real(86401.1, r4_kind)) .ne. real_to_time_type(real(86401.1, r8_kind))) then
```

Comment:
do you think it would make a difference if using 86401.1_r4_kind and  86401.1_r8_kind?